org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/WProcessCtl.java

```diff
@@ -116,7 +116,8 @@ public static void process (int WindowNo, ProcessInfo pi, Trx trx, EventListener
 				para.addEventListener(DialogEvents.ON_WINDOW_CLOSE, new EventListener<Event>() {
 					@Override
 					public void onEvent(Event event) throws Exception {
-						parent.hideMask();
+						if(para.getDesktop() != null)
+							parent.hideMask();
```

Comment:
in case you miss it, this looks wrong to me, can you post the stack trace that support the need of this ?

Reply:
hi @hengsin, the stack trace without this check:

```
java.lang.NullPointerException
	at org.zkoss.zk.ui.http.ExecutionImpl$1.getValue(ExecutionImpl.java:112)
	at org.zkoss.web.servlet.xel.StringKeysMap.get(StringKeysMap.java:38)
	at org.zkoss.zk.ui.AbstractComponent.getAttribute(AbstractComponent.java:925)
	at org.adempiere.webui.ShowMaskWrapper.hasFlagShowMask(ShowMaskWrapper.java:125)
	at org.adempiere.webui.ShowMaskWrapper.hideMask(ShowMaskWrapper.java:77)
	at org.adempiere.webui.component.Window.hideMask(Window.java:137)
	at org.adempiere.webui.apps.WProcessCtl$1.onEvent(WProcessCtl.java:120)
	at org.zkoss.zk.ui.AbstractComponent.onEvent(AbstractComponent.java:3191)
	at org.zkoss.zk.ui.AbstractComponent.service(AbstractComponent.java:3161)
	at org.zkoss.zk.ui.AbstractComponent.service(AbstractComponent.java:3103)
	at org.zkoss.zk.ui.impl.EventProcessor.process(EventProcessor.java:138)
	at org.zkoss.zk.ui.event.Events.sendEvent(Events.java:367)
	at org.adempiere.webui.component.Window.onPageDetached(Window.java:93)
	at org.adempiere.webui.apps.ProcessModalDialog.onPageDetached(ProcessModalDialog.java:308)
	at org.zkoss.zk.ui.AbstractComponent.setPage0(AbstractComponent.java:520)
	at org.zkoss.zk.ui.AbstractComponent.setPage0(AbstractComponent.java:525)
	at org.zkoss.zk.ui.AbstractComponent.setPage0(AbstractComponent.java:525)
	at org.zkoss.zk.ui.AbstractComponent.setPage0(AbstractComponent.java:525)
	at org.zkoss.zk.ui.AbstractComponent.setPage0(AbstractComponent.java:525)
	at org.zkoss.zk.ui.AbstractComponent.setPage0(AbstractComponent.java:525)
	at org.zkoss.zk.ui.AbstractComponent.setPage0(AbstractComponent.java:525)
```

Reply:
I see, thank you, I've tried and committed it

Reply:
hi @PeterTakacs300 , actually I means to remove the changes from here and add the native request check to ShowMaskWrapper.setFlagShowMask and setFlagShowMask.hasFlagShowMask method. Can you test whether that works for your use case here as well ?

Reply:
hi @hengsin, ok I added an if condition to setFlagShowMask and an "&&" logic to hasFlagShowMask so it returns false if `Executions.getCurrent() != null && Executions.getCurrent().getNativeRequest() != null` is false - looks like this avoids the error (latest commit), is it correct this way?

Reply:
yes, looks good. I can approve this if you have tested it for the issue reported.

Reply:
yes it's tested, fixes the reported issue